x-pack/solutions/observability/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/badge/span_links_badge.tsx

```diff
@@ -45,13 +45,17 @@ export function SpanLinksBadge({ linkedParents, linkedChildren, id, onClick }: P
     >
       <EuiBadge
         data-test-subj={`spanLinksBadge_${id}`}
-        onClick={(e: any) => {
-          e.stopPropagation();
-          onClick(spanLinksFlyoutTab);
-        }}
-        onClickAriaLabel={i18n.translate('xpack.apm.waterfall.spanLinks.badgeAriaLabel', {
-          defaultMessage: 'Open span links details',
-        })}
+        {...(onClick
+          ? {
+              onClick: (e: any) => {
+                e.stopPropagation();
+                onClick(spanLinksFlyoutTab);
+              },
+              onClickAriaLabel: i18n.translate('xpack.apm.waterfall.spanLinks.badgeAriaLabel', {
+                defaultMessage: 'Open span links details',
+              }),
+            }
+          : [])}
```

Comment:
Alternatively. (just a suggestion, no strong preference here)

```suggestion
          onClick={
            onClick
              ? (e) => {
                  e.stopPropagation();
                  onClick(spanLinksFlyoutTab);
                }
              : () => {}
          }
          onClickAriaLabel={
            onClick
              ? i18n.translate('xpack.apm.waterfall.spanLinks.badgeAriaLabel', {
                  defaultMessage: 'Open span links details',
                })
              : undefined
          }
```